File: Cosmos-IOS/Cosmostation/Cosmostation/Model/KavaModel/ClaimMultiplier.swift
//
//  ClaimMultiplier.swift
//  Cosmostation
//
//  Created by 정용주 on 2021/03/04.
//  Copyright © 2021 wannabit. All rights reserved.
//

import Foundation

public struct ClaimMultiplier {
    var denom: String?
    var multipliers: Array<Multiplier> = Array<Multiplier>()
    
    init(_ dictionary: NSDictionary?) {
        self.denom = dictionary?["denom"] as? String
        if let rawMultipliers = dictionary?["multipliers"] as? Array<NSDictionary>  {
            for rawMultiplier in rawMultipliers {
                self.multipliers.append(Multiplier(rawMultiplier))
            }
        }
    }
}

public struct Multiplier {
    var name: String?
    var months_lockup: String?
    var factor: NSDecimalNumber = NSDecimalNumber.zero
    
    init(_ dictionary: NSDictionary?) {
        self.name = dictionary?["name"] as? String
        self.months_lockup = dictionary?["months_lockup"] as? String
        if let rawFactor = dictionary?["factor"] as? String {
            self.factor = NSDecimalNumber.init(string: rawFactor)
        }
    }
}

File: Cosmos-Android/app/src/main/java/wannabit/io/cosmostaion/model/kava/ClaimMultiplier.java
package wannabit.io.cosmostaion.model.kava;

import com.google.gson.annotations.SerializedName;

import java.util.ArrayList;

public class ClaimMultiplier {
    @SerializedName("denom")
    public String denom;

    @SerializedName("multipliers")
    public ArrayList<Multiplier> multipliers;

    public class Multiplier {
        @SerializedName("name")
        public String name;

        @SerializedName("months_lockup")
        public String months_lockup;

        @SerializedName("factor")
        public String factor;
    }
}
